File: specification/network/resource-manager/Microsoft.Network/stable/2025-03-01/examples-java/VpnConnectionStartPacketCapture.java

import com.azure.resourcemanager.network.models.VpnConnectionPacketCaptureStartParameters;
import java.util.Arrays;

/**
 * Samples for VpnConnections StartPacketCapture.
 */
public final class Main {
    /*
     * x-ms-original-file: specification/network/resource-manager/Microsoft.Network/stable/2025-03-01/examples/
     * VpnConnectionStartPacketCapture.json
     */
    /**
     * Sample code: Start packet capture on vpn connection without filter.
     * 
     * @param azure The entry point for accessing resource management APIs in Azure.
     */
    public static void
        startPacketCaptureOnVpnConnectionWithoutFilter(com.azure.resourcemanager.AzureResourceManager azure) {
        azure.networks().manager().serviceClient().getVpnConnections()
            .startPacketCapture(
                "rg1", "gateway1", "vpnConnection1", new VpnConnectionPacketCaptureStartParameters()
                    .withLinkConnectionNames(Arrays.asList("siteLink1", "siteLink2")),
                com.azure.core.util.Context.NONE);
    }
}

File: specification/network/resource-manager/Microsoft.Network/Network/stable/2025-07-01/examples-js/VpnConnectionStartPacketCapture.js
const { NetworkManagementClient } = require("@azure/arm-network");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to starts packet capture on Vpn connection in the specified resource group.
 *
 * @summary starts packet capture on Vpn connection in the specified resource group.
 * x-ms-original-file: 2025-07-01/VpnConnectionStartPacketCapture.json
 */
async function startPacketCaptureOnVpnConnectionWithoutFilter() {
  const credential = new DefaultAzureCredential();
  const subscriptionId = "00000000-0000-0000-0000-000000000000";
  const client = new NetworkManagementClient(credential, subscriptionId);
  const result = await client.vpnConnections.startPacketCapture(
    "rg1",
    "gateway1",
    "vpnConnection1",
    { parameters: { linkConnectionNames: ["siteLink1", "siteLink2"] } },
  );
  console.log(result);
}
